qutebrowser/browser/network/networkmanager.py

```diff
@@ -358,6 +358,23 @@ def createRequest(self, op, req, outgoing_data):
             dnt = '0'.encode('ascii')
         req.setRawHeader('DNT'.encode('ascii'), dnt)
         req.setRawHeader('X-Do-Not-Track'.encode('ascii'), dnt)
+
+        def same_domain():
+            tabbed_browser = objreg.get('tabbed-browser', scope='window', window=self._win_id)
+            view = tabbed_browser.currentWidget()
+            print(req.url().host() == view.url().host(), repr(view.url().host()), repr(req.url().host()))
+            # TODO: We probably want to allow headers if we're on test.com and
+            # doing a request to www.test.com? Or maybe a new settings value for
+            # this?
+            return req.url().host() == view.url().host()
```

Comment:
Sorry for the spam - here's the [changes between the two lists](http://paste.the-compiler.org/view/a1aafa15) (from 5.2 to the current master).

Seems like *mostly* new TLDs were added - maybe use Qt's `topLevelDomain` and if it returns an empty string, just assume everything until the last part is a TLD? This will yield a few false-positives when looking at the diff, but I guess that's okay for now.